### PR TITLE
Add remote

### PR DESCRIPTION
**Pull request by npentrel**

### CODE REVIEW

File: sensor/Makefile [hunk 1]
```diff
@@ -2,6 +2,9 @@
 bin/customsensor: *.go cmd/module/*.go
 	go build -o bin/customsensor cmd/module/cmd.go
 
+remote:
```
Comment (Otterverse) [inline sensor/Makefile]: This should probably be similar to the one above. E.g. a real file as the target, and deps on the go files involved, then just "remote" becomes a target that depends on the real-file target.

File: sensor/Instructions.md [hunk 2]
```diff
@@ -51,14 +51,25 @@ make updaterdk
...
 
-#### 6. Test your module locally
+#### 5. Test your module locally
 
-Follow the instructions to [Add a local module](https://docs.viam.com/registry/configure/#add-a-local-module).
+While testing, we recommend that you add the resources you've built with a remote to your machine because it allows you to recompile your code without having to restart your entire machine:
 
-#### 7. Upload your module to the registry
+1. Edit the [`cmd/module/cmd.go` file](./cmd/module/cmd.go). The code there should configure and run a robot with your modular resource configured.
+2. If you haven't already got a machine, create a new machine on [the Viam app](app.viam.com) and follow the setup instructions on the setup tab.
```
Comment (Otterverse) [inline sensor/Instructions.md]:
```suggestion
2. If you don't already have a machine, create a new machine on [the Viam app](app.viam.com) and follow the setup instructions.
```
Setup is no longer a "tab" with R2D2

File: sensor/Instructions.md [hunk 3]
```diff
@@ -51,14 +51,25 @@ make updaterdk
...
 Next, run `make bin/customsensor` to build the executable.
 
-#### 6. Test your module locally
+#### 5. Test your module locally
 
-Follow the instructions to [Add a local module](https://docs.viam.com/registry/configure/#add-a-local-module).
+While testing, we recommend that you add the resources you've built with a remote to your machine because it allows you to recompile your code without having to restart your entire machine:
 
-#### 7. Upload your module to the registry
+1. Edit the [`cmd/module/cmd.go` file](./cmd/module/cmd.go). The code there should configure and run a robot with your modular resource configured.
```
Comment (Otterverse) [inline sensor/Instructions.md]: I don't understand this instruction. The cmd/module is supposed to be a module. It's not clear at all what it means for the "code to configure and run a robot" (which is normally something humans do via App.)

File: sensor/Instructions.md [hunk 4]
```diff
@@ -51,14 +51,25 @@ make updaterdk
...
+2. If you haven't already got a machine, create a new machine on [the Viam app](app.viam.com) and follow the setup instructions on the setup tab.
+3. Run `make remote` to compile the [`cmd/module/cmd.go` file](./cmd/module/cmd.go).
```
Comment (Otterverse) [inline sensor/Instructions.md]: `make remote` doesn't compile the module... `make bin/customsensor` does.

File: sensor/Instructions.md [hunk 5]
```diff
@@ -51,14 +51,25 @@ make updaterdk
...
+4. Run `./bin/remoteserver my_sensor_name` and specify any additional commandline arguments that your code passes  to the modular resource attributes. This will instantiate a simplified instance of `viam-server` with your modular resource.
```
Comment (Otterverse) [inline sensor/Instructions.md]: There are no instructions/TODOs in cmd/remote to explain how CLI arguments are used or consumed.

File: sensor/Instructions.md (hunk 2 above)
Comment (Otterverse) [reply sensor/Instructions.md]: Secondly, why are you having them create a machine if they're making a remote just for testing. If they're going to actually run viam-server (per the setup instructions) then they can just configure their module, and not mess with a remote at all. The reason a "remote" makes testing easier is you can directly use the local control (web) page without having to configure a real robot or mess with App at all.

File: sensor/Instructions.md [hunk 6]
```diff
@@ -51,14 +51,25 @@ make updaterdk
...
+5. On the machine on the Viam app, go to your **Config** tab and to the **Remotes** subtab. Name your remote `remotesensor` and click **Create remote**, then add `localhost:8083` to the **Address** field.
+6. Click **Save config**
+7. Navigate to your **Control** tab on the Viam app. If everything is connected and working, you will see your modular resource and be able to test it.
```
Comment (Otterverse) [inline sensor/Instructions.md]: [nit] It's not a "modular" resource if it's a remote.

File: sensor/Instructions.md
Comment (Otterverse) [inline sensor/Instructions.md:1]: Line 29 points readers to a file that doesn't exist.

**Review by Otterverse — CHANGES_REQUESTED** [review]
Lots of inline comments. Overall, I don't think it's clear at all how to make correct use of the new cmd/remote bit (or how to modify it.)

File: sensor/Instructions.md (hunk 2 above)
Comment (npentrel) [reply sensor/Instructions.md]: When I tried this the local control page at localhost:8083 did not show the sensor. It only shows operations and sessions
Screenshot 2024-03-22 at 17 49 21

Should it show the sensor there?

File: sensor/Makefile (hunk 1 above)
Comment (npentrel) [reply sensor/Makefile]: I think I've fixed this now

File: sensor/Instructions.md (hunk 3 above)
Comment (npentrel) [reply sensor/Instructions.md]: it's supposed to reference the cmd/remote/cmd.go file. Sorry. Yeah - I suppose I need to speak to Eliot why he prefers testing this way.

File: sensor/Instructions.md (hunk 4 above)
Comment (npentrel) [reply sensor/Instructions.md]: this should all say `cmd/remote/cmd.go`

File: sensor/Instructions.md (hunk 6 above)
Comment (npentrel) [reply sensor/Instructions.md]: that's interesting. I think of a modular resource as anything that isn't a builtin resource. How would you refer to the configured component on the remote now?

**Pull request merged**